src/pykiso/lib/auxiliaries/communication_auxiliary.py

```diff
@@ -47,7 +50,45 @@ def send_message(self, raw_msg: bytes) -> bool:
 
         :return: True if command was executed otherwise False
         """
-        return self.run_command("send", raw_msg, timeout_in_s=None)
+        return self.run_command("send", raw_msg)
+
+    def run_command(
+        self,
+        cmd_message: Any,
+        cmd_data: Any = None,
+        blocking: bool = True,
+        timeout_in_s: int = None,
+    ) -> bool:
+        """Send a request by transmitting it through queue_in and
+        populate queue_tx with the command verdict (successful or not).
+
+        :param cmd_message: command to send
+        :param cmd_data: data you would like to populate the command
+            with
+        :param blocking: If you want the command request to be
+            blocking or not
+        :param timeout_in_s: Number of time (in s) you want to wait
+            for an answer
+
+        :return: True if the request is correctly executed otherwise
+            False
+        """
+        with self.lock:
+            log.debug(
+                f"sending command '{cmd_message}' with payload {cmd_data} using {self.name} aux."
+            )
+            state = None
+            self.queue_in.put((cmd_message, cmd_data))
+            try:
+                state = self.queue_tx.get(blocking, timeout_in_s)
+                log.debug(
+                    f"command '{cmd_message}' successfully sent for {self.name} aux"
+                )
+            except queue.Empty:
```

Comment:
Same here

Reply:
blocking=False , this mean if something is not directly in the queue the exception queueEmpty is raised so it advertise you that you expected soething directly but nothing is in the queue

Reply:
If I said blocking=False, why should I get an error? You just asking to not wait for the queue to get populate just, just return what is inside the queue. If the queue is empty you get the exception queue.Empty.